packages/jaeger-ui/src/components/TracePage/CriticalPath/utils/sanitizeOverFlowingChildren.tsx

```diff
@@ -27,80 +27,71 @@ const sanitizeOverFlowingChildren = (spanMap: Map<string, Span>): Map<string, Sp
 
   spanIds.forEach(spanId => {
     const span = spanMap.get(spanId)!;
-    if (span && span.references.length) {
-      // parentSpan will be undefined when its parentSpan is dropped previously
-      const parentSpan = spanMap.get(span.references[0].spanID);
-      if (parentSpan) {
-        const childEndTime = span.startTime + span.duration;
-        const parentEndTime = parentSpan.startTime + parentSpan.duration;
-        switch (true) {
-          case span.startTime >= parentSpan.startTime && childEndTime <= parentEndTime:
-            // case 1: everything looks good
-            // |----parent----|
-            //   |----child--|
-            break;
+    if (!(span && span.references.length)) {
+      return;
+    }
+    // parentSpan will be undefined when its parentSpan is dropped previously
+    const parentSpan = spanMap.get(span.references[0].spanID);
+    if (parentSpan) {
+      const childEndTime = span.startTime + span.duration;
+      const parentEndTime = parentSpan.startTime + parentSpan.duration;
+      if (span.startTime >= parentSpan.startTime) {
+        if (span.startTime >= parentEndTime) {
+          // child outside of parent range => drop the child span
+          //      |----parent----|
+          //                        |----child--|
+          // Remove the childSpan from spanMap
+          spanMap.delete(span.spanID);
 
-          case span.startTime < parentSpan.startTime &&
-            childEndTime <= parentEndTime &&
-            childEndTime > parentSpan.startTime:
-            // case 2: child start before parent, truncate is needed
+          // Remove the childSpanId from its parent span
+          parentSpan.childSpanIds = parentSpan.childSpanIds.filter(id => id === span.spanID);
```

Comment:
`filter` function keeps the items if the condition is true. The result of this line will be `childSpanIds == [span.spanOD]`, is that what you wanted?

Reply:
Yeah I think I did a typo here. I need result to be all child spanIds except span.spanID

Reply:
so is this a bug? should it be caught by a unit test?